core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/BackupRestoreDBStateParamsProvider.java

```diff
@@ -48,17 +50,19 @@ public Map<String, Object> createParamsForBackupRestore(String backupLocation) {
         upgradeParams.put(EMBEDDED_DB_PORT_KEY, "5432");
         upgradeParams.put(EMBEDDED_DB_USER_KEY, "postgres");
         upgradeParams.put(EMBEDDED_DB_PASSWORD_KEY, "postgres");
-        setCloudStorageBackupParameters(backupLocation, upgradeParams);
+        setCloudStorageBackupParameters(backupLocation, upgradeParams, backupInstanceProfile);
+        LOGGER.debug("Created parameters for RDS upgrade backup and restore: {}", upgradeParams);
         return params;
     }
 
-    private void setCloudStorageBackupParameters(String backupLocation, Map<String, String> upgradeParams) {
+    private void setCloudStorageBackupParameters(String backupLocation, Map<String, String> upgradeParams, String backupInstanceProfile) {
         if (StringUtils.isNotBlank(backupLocation) && backupLocation.startsWith(FileSystemType.ADLS_GEN_2.getProtocol())) {
             AdlsGen2Config adlsGen2Config = adlsGen2ConfigGenerator.generateStorageConfig(backupLocation);
             upgradeParams.put(BACKUP_LOCATION_KEY, backupLocation);
             upgradeParams.put(ABFS_ACCOUNT_NAME_KEY, adlsGen2Config.getAccount());
             upgradeParams.put(ABFS_FILE_SYSTEM_KEY, adlsGen2Config.getFileSystem());
             upgradeParams.put(ABFS_FILE_SYSTEM_FOLDER_KEY, adlsGen2Config.getFolderPrefix());
+            upgradeParams.put(BACKUP_INSTANCE_PROFILE, backupInstanceProfile);
```

Comment:
No. However, is it possible to have null for the instance profile? I guess there should always be a logger profile, otherwise the env could not work. What use-case do you have in mind?

Reply:
Hm yeah, you are right, that's not a valid scenario.